app/services/fees/create_pay_in_advance_service.rb

```diff
@@ -8,7 +8,12 @@ def initialize(charge:, event:, billing_at: nil, estimate: false)
       @charge = charge
       @event = Events::CommonFactory.new_instance(source: event)
       @billing_at = billing_at || @event.timestamp
-      @estimate = estimate
+
+      @estimate = if @event.persisted
+        estimate
+      else
+        true
+      end
```

Comment:
I feel like it would be must more explicit and easier to understand if we had:

```suggestion
      raise ArgumentError, "estimate must be true if event if not persisted" if !event.persisted? && !estimate
```

And then pass `estimate: true` in the caller.